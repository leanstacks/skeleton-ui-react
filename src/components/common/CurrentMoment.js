import React from 'react';
import { DateTime } from 'luxon';

export default class CurrentMoment extends React.Component {

  constructor(props) {
    super(props);

    const { format = 'f' } = this.props;
    const now = DateTime.now().toFormat(format);

    this.state = {
      format,
      now
    };
  }

  componentDidMount() {
    this.intervalId = setInterval(this.refreshCurrentMoment, 5000);
  }

  componentWillUnmount() {
    clearInterval(this.intervalId);
  }

  refreshCurrentMoment = () => {
    const now = DateTime.now().toFormat(this.state.format);
    this.setState({ now });
  };

  render() {
    return (
      <span>{this.state.now}</span>
    );
  }

};