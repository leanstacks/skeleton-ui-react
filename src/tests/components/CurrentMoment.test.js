import React from 'react';
import { shallow } from 'enzyme';
import { DateTime, Settings } from 'luxon';

import CurrentMoment from '../../components/common/CurrentMoment';

beforeEach(() => {
  Settings.now = () => 0;
});

test('should render CurrentMoment correctly', () => {
  const wrapper = shallow(<CurrentMoment />);
  expect(wrapper).toMatchSnapshot();
});

test('should render CurrentMoment with format correctly', () => {
  const wrapper = shallow(<CurrentMoment format='YYYY' />);
  expect(wrapper).toMatchSnapshot();
});