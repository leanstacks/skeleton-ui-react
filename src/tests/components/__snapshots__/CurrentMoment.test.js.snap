// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CurrentMoment correctly 1`] = `
<span>
  12/31/1969, 7:00 PM
</span>
`;

exports[`should render CurrentMoment with format correctly 1`] = `
<span>
  YYYY
</span>
`;
